environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/DiscoveryStrategyFactory.java

```diff
@@ -42,20 +47,31 @@ private DiscoveryStrategyFactory() {
      */
     public static DiscoveryStrategy create(ResourceLoader resourceLoader, Bootstrap bootstrap,
         Set<Class<? extends Annotation>> initialBeanDefiningAnnotations, boolean jandexStrategyDisabled) {
-        if (Jandex.isJandexAvailable(resourceLoader)) {
-            if (jandexStrategyDisabled) {
-                CommonLogger.LOG.jandexDiscoveryStrategyDisabled();
-            } else {
-                CommonLogger.LOG.usingJandex();
-                try {
-                    return Jandex.createJandexDiscoveryStrategy(resourceLoader, bootstrap, initialBeanDefiningAnnotations);
-                } catch (Exception e) {
-                    throw CommonLogger.LOG.unableToInstantiate(Jandex.JANDEX_DISCOVERY_STRATEGY_CLASS_NAME,
-                        Arrays.toString(new Object[] { resourceLoader, bootstrap, initialBeanDefiningAnnotations }), e);
-                }
+        DiscoveryStrategy returnValue = null;
+        final Iterator<Metadata<DiscoveryStrategy>> iterator = ServiceLoader.load(DiscoveryStrategy.class, resourceLoader).iterator();
+        if (iterator != null && iterator.hasNext()) {
+            final DiscoveryStrategy candidate = iterator.next().getValue();
+            if (candidate != null) {
+                candidate.setResourceLoader(resourceLoader);
+                candidate.setBootstrap(bootstrap);
+                candidate.setInitialBeanDefiningAnnotations(initialBeanDefiningAnnotations);
+                returnValue = candidate;
             }
+        } else if (jandexStrategyDisabled) {
+            CommonLogger.LOG.jandexDiscoveryStrategyDisabled();
+        } else if (Jandex.isJandexAvailable(resourceLoader)) {
```

Comment:
You swapped the checks for `jandexStrategyDisabled` and `Jandex.isJandexAvailable(resourceLoader)`.
Please keep them in the original order as the logging is meant to happen only in case Jandex is available but deliberately not being used.

Reply:
Oh!  My mistake!  I'll fix this shortly.